test/e2e/common_test.go

```diff
@@ -556,8 +556,16 @@ func (p *PodmanTestIntegration) Cleanup() {
 
 	p.StopRemoteService()
 	// Nuke tempdir
-	if err := os.RemoveAll(p.TempDir); err != nil {
-		fmt.Printf("%q\n", err)
+	if isRootless() {
+		// If rootless, os.RemoveAll() is failed due to permission denied
+		cmd := exec.Command(p.PodmanBinary, "unshare", "rm", "-rf", p.TempDir)
+		if err := cmd.Run(); err != nil {
+			fmt.Fprintf(os.Stderr, "%v\n", err)
+		}
+	} else {
+		if err := os.RemoveAll(p.TempDir); err != nil {
+			fmt.Printf("%q\n", err)
+		}
```

Comment:
please put this is a function that accepts the path as arg so we do not duplicate the same lines twice